### ISSUE TITLE
Second attempt is made to add keys when using advanced trading with vehicles

### ISSUE THREAD

**Issue by ndavalos** [EpochModTeam/DayZ-Epoch#2006]
In:
DayZ-Epoch/SQF/dayz_code/actions/AdvancedTrading/functions/z_at_buyItems.sqf

This line:
https://github.com/EpochModTeam/DayZ-Epoch/blob/efaf1f93990eba7b8b71be1c7b6c12cd9e95884b/SQF/dayz_code/actions/AdvancedTrading/functions/z_at_buyItems.sqf#L126
will add a key to the player (from epoch_generateKey):
https://github.com/EpochModTeam/DayZ-Epoch/blob/9b96ddd2c3f63a263d2408a0bfe95571fa8f32dd/SQF/dayz_code/compile/epoch_generateKey.sqf#L25

Unfortunately, z_at_buyItems.sqf also attempts to add the same key to the player later here:
https://github.com/EpochModTeam/DayZ-Epoch/blob/efaf1f93990eba7b8b71be1c7b6c12cd9e95884b/SQF/dayz_code/actions/AdvancedTrading/functions/z_at_buyItems.sqf#L177
here:
https://github.com/EpochModTeam/DayZ-Epoch/blob/efaf1f93990eba7b8b71be1c7b6c12cd9e95884b/SQF/dayz_code/actions/AdvancedTrading/functions/z_at_buyItems.sqf#L203
and here:
https://github.com/EpochModTeam/DayZ-Epoch/blob/efaf1f93990eba7b8b71be1c7b6c12cd9e95884b/SQF/dayz_code/actions/AdvancedTrading/functions/z_at_buyItems.sqf#L242

So basically after generatekey is done, you end up duplicating the key again. Generate key may fail when adding it to inventory if there is no room of course, but I would think that it wouldn't happen since you check for room earlier in the process of purchasing (I think). RPT gets a message about duplicate weapon <keyname> if buying into gear.

**Comment by oiad** [comment]
Fixed it, my mistake when I changed it to use epoch_tempKeys
Cheers!
